usagov/usagov-2021
# Usagov 818 add robots noindex to metarefresh page

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-818

## Description

Created a custom module where I subscribe to a KernelEvents::RESPONSE event to add the robots noindex to redirect pages.

Some of the resources that I used:
* https://symfony.com/doc/current/components/http_kernel.html#the-request-response-lifecycle
* https://symfony.com/doc/current/components/http_foundation.html#request
* https://www.drupal.org/docs/develop/creating-modules/subscribe-to-and-dispatch-events#s-drupal-hooks
* https://stackoverflow.com/questions/18205039/is-it-possible-to-change-the-default-redirect-message-in-symfony
* https://drupalize.me/blog/responding-events-drupal
* https://drupalsun.com/philipnorton42/2022/05/22/drupal-9-correct-way-redirect-page#

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


1. Run `bin/drush cim` and run the project
2. Go to Docker Desktop and find the **cms** container
3. Click the 3 dots in the Actions column and click "Open in terminal"
4. In the terminal, type `./scripts/tome-run.sh` and wait for tome to run
5. Once tome is done, look for the `html` folder and find a page that redirects. You can find the redirects here: /admin/config/search/redirect
6. Check that the html of the redirect has this line `<meta name="robots" content="noindex" />`
<img width="1012" alt="image" src="https://github.com/user-attachments/assets/6762f628-d286-4373-a687-ad5485c6d0f6">

7. Now go to several links on the site that do a redirect and check that it works as usual. **Test english and spanish pages.**
8. Now check that the pages that do not redirect work as usual. **Test english and spanish pages.**

### Requires New Config

### Requires New Content

### Validation Steps

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
10. Find the commit of this pull request.
11. Build and deploy the changes.
12. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
